Implement Calculator Logic and Event Handlers

Spidey-Acer opened the pull request:
In this pull request, I added the logic and event handlers to fully make the Calculator component functional. The project requirements were carefully followed, and I ensured that all the general and JavaScript requirements were met.

**General Requirements:**

- All linter errors have been addressed to maintain code quality and consistency.
- Gitflow was strictly adhered to, ensuring a proper branching and merging strategy.
- The project was well-documented in a professional manner, providing clear insights into the implementation.

**JavaScript Requirements:**

- Best practices for JavaScript were followed, adhering to industry standards.

**Project Requirements:**

- I created a new directory named `logic` within the project.
- The files `calculate.js` and `operate.js` were copied into the `logic/` directory.
- The `big.js` library was installed using NPM to handle decimal calculations.
- Detailed analysis was performed on `calculate.js` and `operate.js` to understand their functionality.

**Implementation Details:**

- The `calculate.js` file was successfully imported into the `Calculator.js` component.
- Click event handlers were implemented using the math logic from `calculate.js` and `operate.js`.
- Each calculator button was made functional, performing the appropriate arithmetic operation, including addition, subtraction, multiplication, and division.

**Testing:**

- Extensive testing was conducted in the browser to ensure the Calculator performs math operations accurately and reliably.

With these changes, the Calculator component is now fully functional, and all the buttons work according to their respective functionalities. The project is now ready for review and deployment.

Please review the changes, and if everything looks good, I'll be ready to merge it and proceed to the next step, otherwise point out areas that need changes.

Thank you for your time and consideration!

Spidey-Acer committed:
Creates logic directory and add calculate.js and operate.js

Spidey-Acer committed:
Import calculate.js and operate.js raw code and installed big.js

Spidey-Acer committed:
Add onClick prop with its prop validation on button.js

Spidey-Acer committed:
Keep track of the calculator's state and handle clicks

Spidey-Acer committed:
Modify the button child component to hear events

Spidey-Acer committed:
Fix linter errors

descholar-ceo reviewed (changes requested):
Hi @Peter-Kibet 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 The PR follows the correct git/github flow ✔️ 
👍🏽 The commit messages are descriptive ✔️ 
👍🏽 Linters are installed and the PR passes the checks ✔️ 
👍🏽 Readme file follows the correct microverse's template ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Your project looks nice and everything works fine, however when a user opens the calculator, we are getting these errors in console, would you make sure that whenever you use the `Button` component you pass it the className as it's been marked as reqired or you can just make it optional if it not really needed:
<img width="907" alt="Screenshot 2023-07-19 at 16 42 13" src="https://github.com/Peter-Kibet/Math_Magician_Calculator/assets/56886064/24cee8f2-57b9-4757-b4e5-3b129d07292d">

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Spidey-Acer committed:
Fix missing className prop in Button component

-Modified the propTypes definition in Button.js to make className prop optional.
-Added a default value of an empty string for the className prop using defaultProps.

prolajumokeoni reviewed (approved):
Hi @Peter-Kibet 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- Changes requested has been implemented   ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
